src/irr/asset/CSTLMeshWriter.cpp

```diff
@@ -150,6 +150,42 @@ bool CSTLMeshWriter::writeMeshBinary(io::IWriteFile* file, const asset::ICPUMesh
 	for (uint32_t i=0; i<mesh->getMeshBufferCount(); ++i)
 	{
 		asset::ICPUMeshBuffer* buffer = mesh->getMeshBuffer(i);
+
+		/*
+			pseudo code that has to be applied in binary and ASCII writing
```

Comment:
@devshgraphicsprogramming  @Crisspl  guys please help me
I have no idea how I can determine which buffer corresponds to which attribute
Is there actaully a way to this? Let's say that I would like to get normals buffer
How can I check it in our engine?

Reply:
https://github.com/Crisspl/IrrlichtBAW/blob/master/include/irr/asset/ICPUMeshDataFormatDesc.h#L117
also see how getAttribPointer() works https://github.com/Crisspl/IrrlichtBAW/blob/master/include/irr/asset/ICPUMeshBuffer.h#L185

Reply:
there's a vague convention that some loaders rely on:
- 0 : position (but there's a query for that)
- 1 : UV
- 2 : Color
- 3 : Normal

Not sure if I got 1 and 2 the right way around, see the OBJ loader, its a good reference.

Some formats like `.baw` or `.gltf` don't really give a shit which attribute does what and can save more than the pos+uv+color+normal.

Reply:
also input to function is const so you can't flip in-place on the mesh

Reply:
Thank you guys, I will remember those hints